api/registrations/serializers.py

```diff
@@ -344,6 +344,19 @@ def update(self, registration, validated_data):
                     raise exceptions.PermissionDenied()
             else:
                 raise exceptions.ValidationError('Registrations can only be turned from private to public.')
+        if validated_data.get('retraction') and not validated_data.get('is_retracted'):
+            raise exceptions.ValidationError(
+                'You cannot provide a withdrawal_justification without a concurrent withdrawal request.',
```

Comment:
The message makes it sound like we're checking to make sure the `withdrawal_justification` isn't being included in a payload for a registration that isn't retracted, but it doesn't look like we're actually checking for that here. Maybe 347 should be `if validated_data.get('withdrawal_justification') and not validated_data.get('withdrawn'):`?

Reply:
This is what I was checking [with this test](https://github.com/CenterForOpenScience/osf.io/pull/8868/files#diff-c8a46efbf3b32bd4666a044196962a8aR499) -- the payload looks like: 

```
'data': {
	'attributes': {'withdrawal_justification': 'Not enough oopmh.'}, 
	'type': 'registrations', 'id': u'n6a2d'
}
```
but since the `withdrawal_justification` field is based on the embedded field `retraction.justification` on the model, `validated_data` ends up looking like: 

```
{u'retraction': {u'justification': u'Not enough oopmh.'}}
```
It's the same with `withdrawn`, which is based on the `is_retracted` field, so comes though as `is_retracted` in `validated_data`.